src/Orc.SystemInfo/Wmi/WindowsManagementConnection.cs

```diff
@@ -0,0 +1,105 @@
+﻿namespace Orc.SystemInfo.Wmi
+{
+    using System;
+    using System.Runtime.InteropServices;
+    using Catel;
+    using Catel.Logging;
+    using Orc.SystemInfo.Win32;
+
+    public sealed class WindowsManagementConnection : IDisposable
+    {
+        private const string DefaultLocalRootPath = @"\\.\root\cimv2";
+
+        private static readonly ILog Log = LogManager.GetCurrentClassLogger();
+
+        private readonly object _lock = new object();
+
+        private bool _disposed;
+        private bool _connected;
+        private IWbemServices _wbemServices;
+        private IWbemContext _context = null;
+
+        public void Open()
+        {
+            try
+            {
+                ThrowIfDisposed();
+
+                if (!_connected)
+                {
+                    IWbemLocator locator = new WbemLocator();
+
+                    lock (_lock)
+                    {
+                        if (!_connected)
+                        {
+                            var authLevel = WbemAuthenticationLevel.PacketIntegrity;
+
+                            _wbemServices = locator.ConnectServer(DefaultLocalRootPath, _context);
+                            _wbemServices.SetProxy(WbemImpersonationLevel.Impersonate, authLevel);
+
+                            _connected = true;
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                _context = null;
+                Dispose();
+                Log.Error(ex);
+            }
+        }
+
+        public void Dispose()
+        {
+            ThrowIfDisposed();
+
+            if (_connected)
+            {
+                lock (_lock)
+                {
+                    if (_wbemServices is not null)
+                    {
+                        Marshal.ReleaseComObject(_wbemServices);
+                        _wbemServices = null;
+                    }
+                    _connected = false;
```

Comment:
I would move it into a separate method Disconnect()
and use try/catch inside of it. Because it could potentially ruin disposing process